ctl/src/http/mod.rs

```diff
@@ -1,10 +1,18 @@
 use axum::{routing::post, Router};
 use tracing::info;
 
+use crate::discovery::DiscoveryHandle;
+
+pub mod deployer;
 pub mod worker;
 
-pub async fn run_server() {
-    let app = Router::new().route("/worker/metrics", post(worker::push_metrics));
+pub async fn run_server(discovery_handle: DiscoveryHandle) {
+    let app = Router::new()
+        .route("/worker/metrics", post(worker::push_metrics))
+        .route(
+            "/deploy",
+            post(deployer::deploy).with_state(discovery_handle),
+        );
```

Comment:
All routes should receive the state.

```suggestion
    let app = Router::new()
        .route("/worker/metrics", post(worker::push_metrics))
        .route(
            "/deploy",
            post(deployer::deploy),
        )
        .with_state(discovery_handle);
```

Later on we'll need a struct to encapsulate all state values (something like an `AppState` struct).